rakefile

```diff
@@ -1,13 +1,79 @@
 require 'rototiller'
 require 'rspec/core/rake_task'
+require 'net/http'
+require 'timeout'
+require 'json'
+
+BASE_ABS_URI = 'https://cinext-abs.delivery.puppetlabs.net/api/v2/'
 
 task :default => :performance
 
-desc 'Run Performance setup'
-rototiller_task :performance do |t|
+desc 'Provision systems using ABS (or use already set ABS_RESOURCE_HOSTS) and then execute beaker performance setup and tests'
+rototiller_task :performance do
+  Rake::Task["performance_provision_with_abs"].execute unless ENV['ABS_RESOURCE_HOSTS']
+  Rake::Task["performance_without_provision"].execute
+  Rake::Task["performance_deprovision_with_abs"].execute unless ENV['BEAKER_PRESERVE_HOSTS'] == 'always' || (@beaker_cmd.result.exit_code != 0 && ENV['BEAKER_PRESERVE_HOSTS'] == 'onfail')
```

Comment:
what happens when @beaker_cmd has not been set?

Reply:
Hmm, looks like this would throw an exception and never run, which probably would be the desired behavior. We probably would want to assume a failure if @beaker_cmd has not been set.

Reply:
i guess that's fine.  do you think the user will know why or what task to run first?

Reply:
I tried to update the readme also. I am also updating the descriptions of the tasks to try to be more clear.